### PR TITLE
Remove Rust projects not under active development

### PR DESCRIPTION
**Pull request by glasser**
We are currently moving forward with the original TypeScript query planner rather than
the late-2020 Rust query planner. We also have not actively worked on `stargate`
recently, and `graphql-parser` is only used by those packages. So for the time
being, remove them from git to simplify the repository and speed up CI.  (If we
find these project valuable later we can always restore them from history.)

This is NOT a complete removal of Rust from this repository: we still have
`harmonizer` and `harmonizer-bin` (which package `@apollo/federation` for use by
Rust code such as Rover). (And of course we are still actively developing other
Apollo Rust projects such as Rover. )However, Harmonizer consists mostly of
wrapper code rather than deep logic, and Rust does not need to be executed
in order to do Node-only work any more.

This consists of:
- Deleting the root directories for `stargate`, `query-planner`,
  `graphql-parser`, and `query-planner-wasm`
- Not setting up Rust for the CircleCI Node testing, because it's no longer
  needed to do Node stuff. This reduces the time to run CircleCI jobs from ~4
  minutes to ~1 minute.
- Removed `stargate`-only GH publish action.
- Remove the duplication of rust-slow-test.yml and rust-test.yml, leaving only
  rust-test.yml which is similar to the old rust-slow-test.yml (but without
  wasm-pack)
- Remove the requirement that rustup be installed in order to run npm things
- Remove grcov.sh (coverage seems less interesting for harmonizer than for real
  tests)

We should seriously consider the suggestion in #602 that harmonizer be moved to
a separate repository to allow this one to have no (slow) CI dependencies on
Rust.

### CODE REVIEW

**Comment by glasser** [comment]
In addition to improving the overall CircleCI build time from ~6m to ~3m, the reduction in the amount of code (and use of wasm-pack) reduces the GitHub workflow runtimes from ~15m to ~5m.

**Review by abernix — APPROVED** [review]
LGTM.

Re: `harmonizer`, since there are advantages to having two packages which are meant to be released in lockstep (one's code is literally packaged inside the other) and would ideally be tested together, I do want to hold off on moving `harmonizer` out (glad you left that out of this PR), but I can definitely see how that change is appealing.  I think we can mitigate the CI time through caching because leveraging incremental compilation it takes about 15 seconds on my computer.  Furthermore, separate repository for what is meant to be a shortish-lived package has some overhead to setting it up that I'd _prefer_ avoiding for the moment.

**Review by martijnwalraven — APPROVED** [review]

File: .github/workflows/rust-test.yml
Comment (abernix) [inline .github/workflows/rust-test.yml:72]: @glasser I don't understand why we would ever fallback to a cache that isn't seeded with our cache-buster value, but happy to continue this after this lands.

Comment (glasser) [reply .github/workflows/rust-test.yml:72]: This was copied from the slow test file.

Comment (abernix) [reply .github/workflows/rust-test.yml:72]: Gotcha. Thanks!

I had tried to remove these having encountered them previously and questioning them and thought you were re-introducing on account of that but this sounds more like the very real perils of the multiple nearly identical but subtly varying confits we had before.  So I'm glad we're whittling those down!  Might still propose removing those but I've also employed a bigger hammer in https://github.com/apollographql/federation/pull/649, should it become necessary in the short term.

**Pull request merged**
